android/src/main/java/com/oney/WebRTCModule/WebRTCModule.java

```diff
@@ -445,270 +449,301 @@ private MediaConstraints parseMediaConstraints(ReadableMap constraints) {
     }
 
     /**
+     * Retrieves "facingMode" constraint value.
+     * @param mediaConstraints a <tt>ReadableMap</tt> which represents "GUM" constraints argument.
+     * @return String value of "facingMode" constraints in "GUM" or <tt>null</tt> if not specified.
+     */
+    private String getFacingMode(ReadableMap mediaConstraints) {
+        if (mediaConstraints != null) {
+            return ReactBridgeUtil.getMapStrValue(mediaConstraints, "facingMode");
+        }
+        return null;
+    }
+
+    /**
      * Retreives "sourceId" constraint value.
      * @param mediaConstraints a <tt>ReadableMap</tt> which represents "GUM"
      * constraints argument
      * @return String value of "sourceId" optional "GUM" constraint or
      * <tt>null</tt> if not specified in the given map.
      */
     private String getSourceIdConstraint(ReadableMap mediaConstraints) {
-        if (mediaConstraints.hasKey("optional")
-                && mediaConstraints.getType("optional") == ReadableType.Array) {
-            ReadableArray optional = mediaConstraints.getArray("optional");
+        if (mediaConstraints != null
+            && mediaConstraints.hasKey("optional")
+            && mediaConstraints.getType("optional") == ReadableType.Array) {
 
+            ReadableArray optional = mediaConstraints.getArray("optional");
             for (int i = 0, size = optional.size(); i < size; i++) {
                 if (optional.getType(i) == ReadableType.Map) {
                     ReadableMap option = optional.getMap(i);
 
                     if (option.hasKey("sourceId")
-                            && option.getType("sourceId") == ReadableType.String) {
+                        && option.getType("sourceId") == ReadableType.String) {
+
                         return option.getString("sourceId");
                     }
                 }
             }
         }
+
         return null;
     }
 
     @ReactMethod
-    public void getUserMedia(ReadableMap constraints,
-                             Callback    successCallback,
-                             Callback    errorCallback) {
-        AudioTrack audioTrack = null;
-        VideoTrack videoTrack = null;
-        WritableArray tracks = Arguments.createArray();
+    public void getUserMedia(final ReadableMap constraints,
+                             final Callback    successCallback,
+                             final Callback    errorCallback) {
```

Comment:
The number of lines within this method is just too big and that hinders readability, understanding, and maintenance. I think it should've been broken into smaller methods. Moreover, the whole `getUserMedia` implementation along with its supporting methods is getting bigger and bigger so I believe it should be extracted out of WebRTCModule.